Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityInputSystemProfileInspector.cs

```diff
@@ -263,8 +267,14 @@ private void RenderList(SerializedProperty list)
                                 if (EditorGUI.EndChangeCheck())
                                 {
                                     serializedObject.ApplyModifiedProperties();
-                                    System.Type type = ((MixedRealityInputSystemProfile)serializedObject.targetObject).DataProviderConfigurations[i].ComponentType.Type;
-                                    ApplyDataProviderConfiguration(type, providerName, configurationProfile, runtimePlatform);
+                                    systemType = ((MixedRealityInputSystemProfile)serializedObject.targetObject).DataProviderConfigurations[i].ComponentType;
+                                    if (systemType != null)
+                                    {
+                                        typeName = systemType.Type?.Name;
+                                        providerAttribute = typeName != null ? GetTypeNameAndProvider(systemType) : null;
+                                    }
```

Comment:
This code is repeated over and over. We don't want redundant code when it can be put in a function. 

Also there should be no need to access the providerAttribute and string of the Type here. 

Keep the original design and just pass the Type to the ApplyDataProviderConfiguration

Reply:
> This code is repeated over and over. We don't want redundant code when it can be put in a function

I totally agree, but the portion that is redundant is much bigger than just my addition to it and slightly different in every derivate of BaseMixedRealityToolkitConfigurationProfileInspector. Do you suggest extracting all of that out first, or is it ok to take my changes in and then do it?